operations/image_processing/subSectionToImage/operation.go

```diff
@@ -0,0 +1,113 @@
+package subsectiontoimage
+
+import (
+	// "bytes"
+
+	"bytes"
+	"fmt"
+	"image"
+	"image/color"
+	"image/jpeg"
+	"image/png"
+	"os"
+
+	"github.com/project-flogo/catalystml-flogo/action/operation"
+	"github.com/project-flogo/core/data/mapper"
+	"github.com/project-flogo/core/data/metadata"
+	"github.com/project-flogo/core/support/log"
+)
+
+type Operation struct {
+	params *Params
+	logger log.Logger
+}
+
+func New(ctx operation.InitContext) (operation.Operation, error) {
+	p := &Params{}
+
+	err := metadata.MapToStruct(ctx.Params(), p, true)
+
+	if err != nil {
+		return nil, err
+	}
+
+	return &Operation{params: p, logger: ctx.Logger()}, nil
+}
+
+type initContextImpl struct {
+	params   map[string]interface{}
+	mFactory mapper.Factory
+	name     string
+}
+
+func (ctx *initContextImpl) Params() map[string]interface{} {
+	return ctx.params
+}
+
+func (ctx *initContextImpl) MapperFactory() mapper.Factory {
+	return ctx.mFactory
+}
+
+func (ctx *initContextImpl) Logger() log.Logger {
+	return log.ChildLogger(log.RootLogger(), ctx.name)
+}
+
+func (a *Operation) Eval(inputs map[string]interface{}) (interface{}, error) {
+	//To get the inputs in the desired types.
+	input := &Input{}
+	input.FromMap(inputs)
+
+	wsub := a.params.Size[0]
+	hsub := a.params.Size[1]
+	xsub := a.params.LowerLeftCorner[0]
+	ysub := a.params.LowerLeftCorner[1]
+
+	// a.logger.Debug("inputs", inputs)
+	a.logger.Info("Starting Operation subsectiontoimage.")
+	a.logger.Debugf("The subection is %dx%d in size with a lower left corner at (%d,%d)", wsub, hsub, xsub, ysub)
+
+	src := input.Img.(image.Image)
+
+	bnds := src.Bounds()
+	fmt.Println(bnds)
```

Comment:
Should this be in log or removed all-together?

Reply:
done